app/models/vote.rb

```diff
@@ -0,0 +1,7 @@
+class Vote < ActiveRecord::Base
+  belongs_to :votable, polymorphic: true
+  validates :votable_id, :votable_type, :user_id, :value, presence: true
+  validates :votable_id, uniqueness: { scope: :user_id }
+  validates :user_id, uniqueness: { scope: :votable_id }
```

Comment:
This second uniqueness validation is redundant since it's validating the exact same thing as defined one line above. Also, both of these identical validations aren't actually validating what you probably want. If you just validate that the user_id/votable_id combination is unique, then you're banning a User from voting on Post with id 42 and Comment with id 42. What you probably want is this:
```ruby
validates :user_id, uniqueness: { scope: [:votable_id, :votable_type] }
```